test/e2e/webdriver/driver.js

```diff
@@ -545,10 +545,19 @@ class Driver {
    * Finds a clickable element on the page using the given locator.
    *
    * @param {string | object} rawLocator - Element locator
-   * @param {number} timeout - Timeout in milliseconds
+   * @param {object} guards
+   * @param {number} [guards.waitAtLeastGuard] - Minimum milliseconds to wait before passing
+   * @param {number} [guards.timeout]  - Timeout in milliseconds
    * @returns {Promise<WebElement>} A promise that resolves to the found clickable element.
    */
-  async findClickableElement(rawLocator, timeout = this.timeout) {
+  async findClickableElement(
+    rawLocator,
+    { waitAtLeastGuard = 0, timeout = this.timeout } = {},
+  ) {
+    assert(timeout > waitAtLeastGuard);
+    if (waitAtLeastGuard > 0) {
+      await this.delay(waitAtLeastGuard);
+    }
```

Comment:
extend the functionality of the method so we can use it with a guard.
Use case: After an action is performed, we want to check whether a button becomes clickable or unclickable.
`waitAtLeastGuard` is default to 0, so there's no change for existing function calls.

Reply:
just wondering, could we achieve this just by passing a custom time out in the above page object? so we don't need to modify the function? :thinking:

Reply:
Oh I think I see what this means: so we want to make sure that the button is not first enabled and then disabled, so we add a buffer time, and then perform the validation. Is that so?

Reply:
yes exactly ! 
Imagine we have a button that is clickable by default. Then, after performing an action, we want to assert that this button becomes unclickable.
However, since the app takes some time to update the button's state after the action, `findClickableElement`will return the element as clickable immediately, even if a large timeout is set.
So we need a guard to wait until the button's state has actually changed to avoid this flakiness,

Reply:
make sense to me! Nice addition :heart_eyes: